File: src/main/java/de/moritzf/quota/dto/RateLimitDto.java
package de.moritzf.quota.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * DTO describing rate-limit availability and its primary/secondary windows.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record RateLimitDto(
        @JsonProperty("allowed") Boolean allowed,
        @JsonProperty("limit_reached") Boolean limitReached,
        @JsonProperty("primary_window") UsageWindowDto primaryWindow,
        @JsonProperty("secondary_window") UsageWindowDto secondaryWindow
) {
}

File: src/main/kotlin/de/moritzf/quota/openai/dto/RateLimitDto.kt
package de.moritzf.quota.openai.dto

import de.moritzf.quota.openai.UsageWindow
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * DTO describing rate-limit availability and its primary/secondary windows.
 */
@Serializable
data class RateLimitDto(
    val allowed: Boolean? = null,
    @SerialName("limit_reached") val limitReached: Boolean? = null,
    @SerialName("primary_window") val primaryWindow: UsageWindowDto? = null,
    @SerialName("secondary_window") val secondaryWindow: UsageWindowDto? = null,
) {
    fun primaryUsageWindow(): UsageWindow? = primaryWindow?.toUsageWindow()

    fun secondaryUsageWindow(): UsageWindow? = secondaryWindow?.toUsageWindow()
}
